subsys/bluetooth/controller/ll_sw/ull_central_iso.c

```diff
@@ -719,3 +736,42 @@ uint16_t ull_central_iso_cis_offset_get(uint16_t cis_handle, uint32_t *cis_offse
 	cis->central.instant = ull_conn_event_counter(conn) + 3;
 	return cis->central.instant;
 }
+
+static void set_bn_max_pdu(bool framed, uint32_t iso_interval,
+			   uint32_t sdu_interval, uint16_t max_sdu, uint8_t *bn,
+			   uint8_t *max_pdu)
+{
+	uint32_t ceil_f_x_max_sdu;
+	uint16_t max_pdu_bn1;
+	uint32_t max_drift;
+	uint32_t ceil_f;
+
+	if (framed) {
+		/* Framed (From ES-18002):
+		 *   Max_PDU >= ((ceil(F) x 5 + ceil(F x Max_SDU)) / BN) + 2
+		 *   F = (1 + MaxDrift) x ISO_Interval / SDU_Interval
+		 *   SegmentationHeader + TimeOffset = 5 bytes
+		 *   Continuation header = 2 bytes
+		 *   MaxDrift (Max. allowed SDU delivery timing drift) = 100 ppm
+		 */
+		max_drift = ceiling_fraction(SDU_MAX_DRIFT_PPM * sdu_interval, 1000000U);
+		ceil_f = ceiling_fraction(iso_interval + max_drift, sdu_interval);
```

Comment:
The `F` calculation doesn't seem to match the mention calculation: `F = (1 + MaxDrift) x ISO_Interval / SDU_Interval`

Is the result the same when using `ceiling_fraction`?

Reply:
That's because MaxDrift is on the SDU_Interval, so the result would be the SDU maximum drift in us + ISO_Interval. This is what I believe is meant by the formula, which is not official yet (to be in Core 5.4).
I may be mistaken though (I wrote it - Vinayak moved it).

Reply:
Hmm, that's concerning. Based on your comment, it sounds like we are implementing something that is not only publically defined or available, but also something that only applies to a spec version that is not yet adopted and that the controller does not yet support?

Reply:
Well, this is what current EBQ test cases expect, so we work with what we have...